### ISSUE TITLE
Limited maximum output zoom cap

### ISSUE THREAD

**Issue by ivan-resetnikov** [manticorp/unrealheightmap#17]
**TLDR:**
There is a limit on how high-resolution the output map image can be, (In the instance below, it is 15) which in some cases is not enough. But you **can** view the map at higher resolutions, this means that you (in one way or another) can get access to the data.

**Detailed overview:**
The website in question is capable of capturing real-world terrain height map data and converting it into usable map images. However, there's a limitation regarding the resolution of the output map image that the website can produce.

In this particular case, the highest resolution for the output map image is set to a maximum value of 15. This means that when you download or export the map image from the website, the highest level of detail you can achieve is capped at this resolution limit.

Despite this restriction, it is important to note that you can still view the map at higher resolutions directly on the website. This implies that while the website's interface allows for viewing more detailed and higher-resolution maps, the output that you can actually save or download does not exceed the resolution level of 15. This limitation might affect the precision and clarity of the terrain data in the saved map image compared to what you can see on the screen.

**My uneducated guess on why that is:**
That's probably because the backend of the map view is not controlled by you, and you are just sending the download requests through some kind of API or interface, which has this limit (to limit the bandwidth or to get developers to pay for a “pro” version that doesn't have the limit).

[Line in code that limits the maximum resolution](https://github.com/manticorp/unrealheightmap/blob/5a2e75c5c4a315e0024506a98c827d680a97d1cc/src/app.ts#L558)

[Issue example link](https://manticorp.github.io/unrealheightmap/#latitude/50.438484037760986/longitude/13.053474426269531/zoom/18/outputzoom/16/width/2048/height/2048)

**Possible solution (for the users):**
(Very rough idea) Write a Python script that will take over your browser for a moment to make screenshots, navigate the terrain, and assemble them into a final image.

Despite this issue, I have to express my gratitude for your work @manticorp. The website is amazing, and I will go as far as saying it is probably the best out there, for the task it is intended to accomplish.

**Comment by manticorp** [comment]
This is purely limited by the source data, from NextZen:

https://www.nextzen.org/

Here's some more info on the heightmap files:

https://www.mapzen.com/blog/terrain-tile-service/

The maximum zoom level is 15 in the source data, which has a resolution of around ~3-10m.

The reason I made this tool was to provide 16 bit grayscale PNGs from this 8 bit PNG encoded source data (the terrarium type files, which encode data across the RGB channels of the image).

There already exist tools which allow you to arbitrarily export these maps in grayscale - this is the project that inspired mine:

https://tangrams.github.io/heightmapper/

It does exactly what you're describing and basically "screenshots" the on screen canvas to get the heightmap at an arbitrary scale, but only with 8 bit depth PNG files. It also does it much faster, because it's able to leverage WebGL rather than me using pure Javascript.

_Why is this bad?_

If you're importing into Unreal Engine, the 8 bit PNG files will have a noticeable stair stepping effect, and the resampling from the arbitrary resolution will give you a very undetailed and 'blurry' result.

I believe my tool currently gives the highest resolution heightmaps for Unreal Engine available, except maybe in certain specific places where higher res heightmaps are available.

As such, I'm going to close this issue, as this is intended in the design, rather than an issue - but thanks for bringing it up 👍
